### PR TITLE
fix(splash-width): added flash mode for heic capturing; added width crop

### PR DESCRIPTION
**Pull request by v-valentynh**
Added flash mode
Added width cropping

### CODE REVIEW

File: ios/RN/RNCamera.m [hunk 1]
```diff
@@ -2367,15 +2372,22 @@ - (void)captureOutput:(AVCapturePhotoOutput *)output didFinishProcessingPhoto:(A
     if (error == nil && self.photoTakenResolve != nil) {
         
         NSData *imageData = [photo fileDataRepresentation];
+        
+        UIImage *takenImage = [UIImage imageWithData:imageData];
+        if (self.options && [self.options[@"width"] integerValue]) {
+            takenImage = [RNImageUtils scaleImage:takenImage toWidth:[self.options[@"width"] integerValue]];
```
Comment (karinfr) [inline ios/RN/RNCamera.m:2383]: nit: we can update imageData to be  tj_UIImageHEICRepresentation(takenImage, 1) here as well?

File: ios/RN/RNCamera.m [hunk 2]
```diff
@@ -864,9 +866,11 @@ - (void)takePicture:(NSDictionary *)options resolve:(RCTPromiseResolveBlock)reso
         
         @try {
             AVCapturePhotoSettings *photoSettings = [AVCapturePhotoSettings photoSettingsWithFormat:@{AVVideoCodecKey : AVVideoCodecTypeHEVC}];
+            [photoSettings setFlashMode:self.flashMode];
```
Comment (karinfr) [inline ios/RN/RNCamera.m]: nice, is flash mode supported only for iOS11? should we also check if isFlashModeSupported: https://developer.apple.com/documentation/avfoundation/avcapturedevice/1386434-isflashmodesupported

**Review by karinfr — APPROVED** [review]
Looks good, thanks for combining flash and compress together

File: ios/RN/RNCamera.m (hunk 1 above)
Comment (v-valentynh) [reply ios/RN/RNCamera.m:2383]: Hey @karinfr 
I believe `tj_UIImageHEICRepresentation` is for `heic` format only.

Comment (karinfr) [reply ios/RN/RNCamera.m:2383]: Yeah, this entire delegate method invoked only for heic. I meant we can move it inside the condition, then we don't need to create `croppedImage` and just to use `imageData`. Just a nit, no action needed:)

**Pull request merged**